### DISCUSSION TITLE
Questions about the runoff concentration and routing mudule

### DISCUSSION THREAD

**Discussion by Aichumei** [iiasa/CWatM#132] [Q&A]
Dear author,
  I reviewed the code and thought the role of the runoff concentration module is to multiply the runoff generations by a coefficient. I didn't find codes to achieve "the runoff generated from each cell is routed to the corner of each cell"，did i leave out some information?
  Also, i found that the channel discharge of each grid cell is contributed by the total runoff of each grid. I did not see codes to identify whether there is a river in the grid cell. Does this mean that the presence of a river in each grid cell is not important , and it's enough to use kinematic wave equations to calculate "channel discharge" without the specific river identifications?
  I’m not sure if my understanding is correct or I leave out some information.
  Thank you for your reply!

**Comment by dof1985** [comment]
Dear @Aichumei 

Below is a short description of the runoff-channel interaction under CWatM.
_Channel definition_
Every grid-cell on CWatM map has a river, as defined in the [ROUTING] section in the settings file with the following variable: ChanMan, ChanLength, ChanWidth, ChanDepth, and ChanGrad.

_Runoff formation_
Direct runoff in a gridcell forms on different landcovers (represented as a fraction of a gridcell). 
For most land covers, direct (surface) runoff is generated after the first two soil layers become saturated; in sealed areas, direct runoff equals precipitation minus some evaporation losses.

_Runoff to routing_
At each time step the surface runoff is collected together with additional flows (e.g., baseflow, return flows, interflow), and sent into the grid cells channel to be routed in the stream network.

_Runoff concentration_
The runoff concentration module introduces a time lag to the process of runoff delivery from the non-channel part of a grid cell and into the channel.  It is applied to all runoff generated in the cell (directrunoff, interflow, baseflow, leakage from canals, and runoff from glacier melt and rain).
Runoff concentration is than divided into different compartment (up tp 10) in the variable (self.var.runoff_conc).  This process run separately for each land cover type and for glaciers and is done with C++ (file `t5.cpp` in the routing_reservoir folder). Every time step - a fraction of this runoff is assigned back to the `self.var.runoff` variable and sent to the channel, using this line of code.
`self.var.runoff = self.var.runoff_conc[0].copy()`

Best
Dor Fridman
